src/JestProcessManagement/JestProcessManager.ts

```diff
@@ -31,13 +31,15 @@ export class JestProcessManager {
   } = {}): JestProcess {
     if (watchMode !== WatchMode.None && this.runAllTestsFirstInWatchMode) {
       return this.runAllTestsFirst(exitedJestProcess => {
-        this.removeJestProcessReference(exitedJestProcess)
-        const jestProcessInWatchMode = this.run({
-          watchMode: WatchMode.Watch,
-          keepAlive,
-          exitCallback,
-        })
-        exitCallback(exitedJestProcess, jestProcessInWatchMode)
+        if (!exitedJestProcess.stopRequested()) {
+          this.removeJestProcessReference(exitedJestProcess)
+          const jestProcessInWatchMode = this.run({
+            watchMode: WatchMode.Watch,
+            keepAlive,
+            exitCallback,
+          })
+          exitCallback(exitedJestProcess, jestProcessInWatchMode)
+        }
```

Comment:
**Non-blocking / Style**:
I'd prefer if we swapped out the `if (!` with a guard clause to explicitly communicate what happens when `stopRequested` is truthy. It's a bit more verbose, but it helps communicate what we're trying to do.
```js
        // When the user input caused the process to stop, stop.
        if (exitedJestProcess.stopRequested()) {
          return;
        }
```